cmd/main.go

```diff
@@ -1,30 +1,34 @@
 package main
 
 import (
+	"flag"
 	"fmt"
-	"io/ioutil"
-	"log"
-	"net/http"
 	"os"
+	"weather"
 )
 
+type Response struct {
+	OneWord string  `json:"oneword"`
+	Celcius float64 `json:"celcius"`
+}
+
 func main() {
 
+	location := flag.String("location", "", "a city")
+	flag.Parse()
+
+	if len(*location) == 0 {
```

Comment:
We have some logic here, don't we? That is to say, there's some behaviour associated with this flag. Specifically, if a location is specified with `-location`, we use that location, but if it isn't, we output an error message and stop.

How could we test this behaviour? Not by calling `main`—we can't do that! Have a think about it and see if anything workable occurs to you.